### PR TITLE
[cmake] A major rewrite of the CMake scripts.

### PR DESCRIPTION
**Pull request by krivenko**
* The root `CMakeLists.txt` has been rewritten almost entirely. Parts of it have been moved into newly created `include/CMakeLists.txt`, `src/CMakeLists.txt` and `doc/CMakeLists.txt`.

* Renamed CMake options `POMEROL_USE_OPENMP` -> `USE_OPENMP` (for shortness) and `POMEROL_BUILD_STATIC` -> `BUILD_SHARED_LIBS` (to follow the standard convention).

* `cmake/CommonDefs.cmake` is gone. Some of the macros defined there were called only once, while the others were never called.

* `pomerolConfig.cmake.in` has been rewritten and moved into `cmake/`.

* `pomerol.pc.in` and `pomerol.lmod.in` have been updated and extended.

* Minor adjustments to the roadmap.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,103 +1,146 @@
 cmake_minimum_required(VERSION 3.1.0)
 
 project(pomerol CXX)
-message("\n${PROJECT_NAME} configuration\n")
-string(TOUPPER ${PROJECT_NAME} PROJECT_CAPNAME)
+set(POMEROL_VERSION 2.0)
+set(POMEROL_DESCRIPTION "An exact diagonalization library aimed at \
+solving condensed matter models of interacting fermions")
+set(POMEROL_URL "http://pomerol.sourceforge.net/")
```
Comment (aeantipov) [inline CMakeLists.txt:7]: Do we want to change this URL btw?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -2,38 +2,37 @@
...
-foreach (test ${tests})
+foreach(test ${tests})
     set(test_src ${test}.cpp)
     add_executable(${test} ${test_src})
     set(test_parameters
         ${MPIEXEC_NUMPROC_FLAG} 1
         ${MPIEXEC_PREFLAGS} ${test} ${MPIEXEC_POSTFLAGS})
     add_test(NAME ${test} COMMAND "${MPIEXEC}" ${test_parameters})
-    target_link_libraries(${test} PRIVATE ${MPI_CXX_LIBRARIES} pomerol catch2)
+    target_link_libraries(${test} ${PROJECT_NAME} ${MPI_CXX_LIBRARIES} catch2)
```
Comment (aeantipov) [inline test/CMakeLists.txt:25]: Out of curiosity, why is PRIVATE gone here?

**Review by aeantipov — APPROVED** [review]
I like the flattened CMakeLists, makes it easier to read/write, thanks!

File: CMakeLists.txt (hunk 1 above)
Comment (krivenko) [reply CMakeLists.txt:7]: Great idea!

Let me keep this URL as it is for now. In a later PR I will be revising the docs and will move them to GitHub Pages (other suggestions?) with automated deployment via GitHub Actions.

File: test/CMakeLists.txt (hunk 2 above)
Comment (krivenko) [reply test/CMakeLists.txt:25]: For no reason, I removed it accidentally.

Strictly speaking, the missing keyword is equivalent to `PUBLIC`, which is not what we want here. In practice, however, no target is going to link to the unit tests, so it does not really matter.

**Pull request merged**